NineChronicles.Headless/GraphTypes/TransactionHeadlessQuery.cs

```diff
@@ -59,6 +66,65 @@ public TransactionHeadlessQuery(StandaloneContext standaloneContext)
                 }
             );
 
+            IEnumerable<Block> ListBlocks(BlockChain chain, long from, long limit)
+            {
+                if (chain.Count <= from)
+                {
+                    yield break;
+                }
+
+                var block = chain[from];
+                for (var i = 0; i < limit; i++)
+                {
+                    if (block is null)
+                    {
+                        yield break;
+                    }
+
+                    yield return block;
+
+                    block = block != chain.Tip ? chain[block.Index + 1] : null;
+                }
+            }
+            Field<ListGraphType<TransactionType>>(
+                name: "ncTransactions",
+                arguments: new QueryArguments(
+                    new QueryArgument<NonNullGraphType<LongGraphType>>
+                        { Name = "startingBlockIndex", Description = "start block index for query tx."},
+                    new QueryArgument<NonNullGraphType<LongGraphType>>
+                        { Name = "limit", Description = "number of block to query."},
+                    new QueryArgument<NonNullGraphType<StringGraphType>>
+                        { Name = "actionType", Description = "filter tx by having actions' type" }
+                ),
+                resolve: context =>
+                {
+                    if (standaloneContext.BlockChain is not BlockChain blockChain)
+                    {
+                         throw new ExecutionError(
+                             $"{nameof(StandaloneContext)}.{nameof(StandaloneContext.BlockChain)} was not set yet!");
+                    }
+
+                    var startingBlockIndex = context.GetArgument<long>("startingBlockIndex");
+                    var limit = context.GetArgument<long>("limit");
+                    var actionType = context.GetArgument<string>("actionType");
+
+                    var blocks = ListBlocks(blockChain, startingBlockIndex, limit);
+                    var transactions = blocks
+                        .SelectMany((block) => block.Transactions)
+                        .Where((tx) => tx.Actions.Any(rawAction =>
+                        {
+                            if (rawAction is not Dictionary action || action["type_id"] is not Text typeId)
+                            {
+                                return false;
+                            }
+
+                            return typeId == actionType;
+                        }));
+
```

Comment:
I think that we can leverage [PLINQ](https://learn.microsoft.com/en-us/dotnet/standard/parallel-programming/introduction-to-plinq) (or any other parallelized way) here because all jobs(i.e., search tx for given action type) against blocks seems to be isolated and don't affect other jobs.